### PR TITLE
Add support for sending messages to Nearby chat from Lua script

### PR DESCRIPTION
**Pull request by maxim-productengine**
Add support for sending messages to Nearby chat via Lua scripts.
Update UI to distinguish between messages sent by the script.

### CODE REVIEW

File: indra/newview/llchathistory.cpp [hunk 1]
```diff
@@ -774,7 +776,7 @@ class LLChatHistoryHeader: public LLPanel
         switch (mSourceType)
         {
             case CHAT_SOURCE_AGENT:
-                icon->setValue(chat.mFromID);
+                icon->setValue(mIsFromScript ? LLSD("Inv_Script") : LLSD(chat.mFromID));
```
Comment (akleshchev) [inline indra/newview/llchathistory.cpp:779]: I feel like having some lua specific id would have been a cleaner option.

**Review by akleshchev — APPROVED** [review]

File: indra/newview/scripts/lua/test_LLChat.lua [hunk 2]
```diff
@@ -0,0 +1,18 @@
+LLChat = require 'LLChat'
+
+function generateRandomWord(length)
+    local alphabet = "abcdefghijklmnopqrstuvwxyz"
+    local word = ""
+    for i = 1, length do
+        local randomIndex = math.random(1, #alphabet)
+        word = word .. alphabet:sub(randomIndex, randomIndex)
```
Comment (nat-goodspeed) [inline indra/newview/scripts/lua/test_LLChat.lua:8]: I realize it doesn't matter for a test script, but when building up a string iteratively, I usually think in terms of appending to a list and then (in Lua) using `table.concat()` to construct the ultimate string.

File: indra/newview/scripts/lua/test_LLChat.lua [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    end
+    return word
+end
+
+local msg = ""
```
Comment (nat-goodspeed) [inline indra/newview/scripts/lua/test_LLChat.lua:13]:
```
local msg = "AI says:"
```
;-)

**Review by nat-goodspeed — APPROVED** [review]
Couple suggestions, but I see you've already merged this.

File: indra/newview/scripts/lua/test_LLChat.lua (hunk 3 above)
Comment (maxim-productengine) [reply indra/newview/scripts/lua/test_LLChat.lua:13]: I'm totally for it!

File: indra/newview/llfloaterimnearbychatlistener.cpp [hunk 4]
```diff
@@ -95,6 +105,6 @@ void LLFloaterIMNearbyChatListener::sendChat(LLSD const & chat_data) const
     }
 
     // Send it as if it was typed in
-    mChatbar.sendChatFromViewer(chat_to_send, type_o_chat, ((BOOL)(channel == 0)) && gSavedSettings.getBOOL("PlayChatAnim"));
+    LLFloaterIMNearbyChat::sendChatFromViewer(chat_to_send, type_o_chat, ((BOOL) (channel == 0)) && gSavedSettings.getBOOL("PlayChatAnim"));
```
[the remark this answers is not in the document]
Comment (maxim-productengine) [reply indra/newview/llfloaterimnearbychatlistener.cpp:108]: Well, I'm glad that BOOLs will be purged.

File: indra/newview/scripts/lua/test_LLChat.lua (hunk 2 above)
Comment (maxim-productengine) [reply indra/newview/scripts/lua/test_LLChat.lua:8]: Sure, why not, let's show 'table' approach too.

**Pull request merged**